store/sqlstore/user_store.go

```diff
@@ -1055,20 +1045,41 @@ func (us SqlUserStore) PermanentDelete(userId string) store.StoreChannel {
 	})
 }
 
-func (us SqlUserStore) AnalyticsUniqueUserCount(teamId string) store.StoreChannel {
+func (us SqlUserStore) Count(options model.UserCountOptions) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
-		query := ""
-		if len(teamId) > 0 {
-			query = "SELECT COUNT(DISTINCT Users.Email) From Users, TeamMembers WHERE TeamMembers.TeamId = :TeamId AND Users.Id = TeamMembers.UserId AND TeamMembers.DeleteAt = 0 AND Users.DeleteAt = 0"
+		query := sq.Select("COUNT(Users.Id)").From("Users")
+
+		if !options.IncludeDeleted {
+			query = query.Where("Users.DeleteAt = 0")
+		}
+
+		if options.IncludeBotAccounts {
+			if options.ExcludeRegularUsers {
+				query = query.LeftJoin("Bots ON Users.Id = Bots.UserId").Where("Bots.UserId IS NOT NULL")
+			}
 		} else {
-			query = "SELECT COUNT(DISTINCT Email) FROM Users WHERE DeleteAt = 0"
+			query = query.LeftJoin("Bots ON Users.Id = Bots.UserId").Where("Bots.UserId IS NULL")
+			if options.ExcludeRegularUsers {
+				// Currenty this doesn't make sense because it will always return 0
+				result.Err = model.NewAppError("SqlUserStore.Count", "UserCountOptions don't make sense", nil, "", http.StatusInternalServerError)
+				return
+			}
+		}
+
+		if options.TeamId != "" {
+			query = query.From("TeamMembers, Users").Where("TeamMembers.TeamId = ? AND Users.Id = TeamMembers.UserId AND TeamMembers.DeleteAt = 0", options.TeamId)
```

Comment:
Perhaps a `JOIN` here would be more explicit in the intent? I typically read more `FROM` conditions as a cross join and have to parse through the `WHERE` clause to figure if it's just a natural join.